### PR TITLE
feat: add project logo api

### PR DESCRIPTION
**Pull request by tomquirk**
per title. used in:
- homepage project logos
- payouts

Can be added in other places later (like activity feed).

Please validate locally!

Notes:
- Cached for 1 day but that may be too long - might want to shorten.
- Some projects with massive logos may fail (cos of Next/vercel API limits)
- Open question: https://github.com/jbx-protocol/juice-interface/pull/3945#discussion_r1284519801

This is an MVP. An optimized solution would:
- replicate (send) images to some image CDN
- the image CDN would auto-downscale/size images 
- we could request whatever size of the image we want
- it would be cached, but we can refresh the cache on-demand, if the project changes their logo.

Screenshot 2023-08-04 at 4 12 25 PM

### CODE REVIEW

File: src/pages/api/juicebox/projectLogo/[projectId].page.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  try {
+    const { projectId } = req.query
+    if (!projectId) {
+      return res.status(400).json({ error: 'projectId is required' })
+    }
+
+    // try getting logo from db. if for some reason its not there, try getting it from chain.
+    const logoUri =
+      (await findLogoFromDb(req, res, projectId as string)) ??
+      (await findLogoOnChain(projectId as string))
```
Comment (tomquirk) [inline src/pages/api/juicebox/projectLogo/[projectId].page.ts:64]: if the project legit doesnt have a logo, then we are doing 2 sync GETS for nothing.

Should we just remove this line and solely rely on the DB?

File: src/pages/api/juicebox/projectLogo/[projectId].page.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * Note, if DB request fails, we still try to fetch the logo from chain. This will be 2 more requests.
+ *
+ * Thus this endpoint should be cached.
+ * However, the ideal duration is TBD.
+ * Please consider changing (lowering) if it causes issues with users.
+ */
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method !== 'GET') {
+    return res.status(404)
```
Comment (wraeth-eth) [inline src/pages/api/juicebox/projectLogo/[projectId].page.ts:52]: method not allowed `405`

Comment (wraeth-eth) [reply src/pages/api/juicebox/projectLogo/[projectId].page.ts:52]: I think you also need to include .json, or something else as well

File: src/pages/api/juicebox/projectLogo/[projectId].page.ts (hunk 1 above)
Comment (wraeth-eth) [reply src/pages/api/juicebox/projectLogo/[projectId].page.ts:64]: I think so 👍

**Review by wraeth-eth — COMMENTED** [review]
Comments - still need to test locally.

Love this @tomquirk , looks really good!!

**Comment by wraeth-eth** [comment]
Just tested! So fresh!

File: src/pages/api/juicebox/projectLogo/[projectId].page.ts (hunk 1 above)
Comment (tomquirk) [reply src/pages/api/juicebox/projectLogo/[projectId].page.ts:64]: rad. will do

File: src/pages/api/juicebox/projectLogo/[projectId].page.ts (hunk 2 above)
Comment (tomquirk) [reply src/pages/api/juicebox/projectLogo/[projectId].page.ts]: great call, done

**Comment by tomquirk** [comment]
Thanks @wraeth-eth for review! <3

**Pull request merged**